### PR TITLE
Add Mailman service to docker compose

### LINKED ISSUE #2781: Discussion regarding the deployment of reply-by-email to comment
As discussed with @jywarren on gitter, opening this issue to discuss the things to be done regarding to start our new `reply-by-email` feature.

### LINKED ISSUE #2878: Need to start mailman server process in order to fetch incoming mail
### Please describe the problem (or idea)

In #2669 we merged functionality for incoming email. We've configured this, but now we need to start an independent process that does the mail fetching. 

@namangupta01 mentioned this:

> We just have to write the configurations in https://github.com/publiclab/plots2/blob/master/script/mailman_server and then we can start the mailman server which is an incoming mail server which will automatically do the work for us.
We only have to write the proper configurations of the email id which will going to receive all the mails.

So now we need to add a container to `containers/docker-compose-*.yml` where we define by code what processes / containers run in staging and production.

@namangupta01  We'll appreciate your help in defining the command that needs to be run - similar to how it was done for Sidekiq (see docker-compose files)

Thanks in advance!

### PR DESCRIPTION
**Pull request by namangupta01**
Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!
Follow up pr of #2669 

Fixes #2781 and fixes #2878

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -39,3 +39,12 @@ services:
       - .:/app
     environment:
       - JOB_WORKER_URL=redis://redis:6379/0
+  mailman:
+    build: .
+    command: cd script && mailman
```
Comment (jywarren) [inline docker-compose.yml]: not `mailman_server`?

Comment (jywarren) [reply docker-compose.yml]: https://github.com/mailman/mailman/blob/master/USER_GUIDE.md

**Comment by jywarren** [comment]
OK, sorry we missed this, Naman! Lots of PRs open right now. Looks amazing. 

We'd like to ask you to move your additions into `/containers/docker-compose-*.yml` -- three versions. These will correspond to the configs we have in staging, unstable, production.

File: docker-compose.yml (hunk 1 above)
Comment (namangupta01) [reply docker-compose.yml]: Yes you are right! 😅. It should be mailman_server.

Comment (namangupta01) [reply docker-compose.yml]: Let me confirm first. Sorry!

**Comment by jywarren** [comment]
Also, the error seems to be:

```
Creating plots2_db_1
ERROR: for plots2_mailman_1  Cannot start service mailman: oci runtime error: container_linux.go:265: starting container process caused "exec: \"cd\": executable file not found in $PATH"
Creating plots2_sidekiq_1
ERROR: for mailman  Cannot start service mailman: oci runtime error: container_linux.go:265: starting container process caused "exec: \"cd\": executable file not found in $PATH"
```

**Comment by namangupta01** [comment]
@jywarren I have changed it. Please review.

**Comment by jywarren** [comment]
Oh! this is interesting!!!

```
ERROR["test_login_user_with_an_email_and_then_connect_google_provider", Minitest::Result, 16.32340962699999]
 test_login_user_with_an_email_and_then_connect_google_provider#Minitest::Result (16.32s)
ArgumentError:         ArgumentError: unknown keyword: user_session
            test/functional/user_sessions_controller_test.rb:78:in `block in <class:UserSessionsControllerTest>'
.#<OmniAuth::AuthHash info=#<OmniAuth::AuthHash::InfoHash email="bansal.sidharth309@gmail.com" name="sidharth bansal"> provider="google_oauth2" uid="1357908642">
.{"uid"=>"98740858591", "info"=>{"email"=>"jeff@gmail.com"}}
.{"uid"=>"98746858591", "info"=>{"email"=>"bansal.sidharth2990@gmail.com"}}



  1) Error:
UserSessionsControllerTest#test_login_user_with_an_email_and_then_connect_google_provider:
ArgumentError: unknown keyword: user_session
    test/functional/user_sessions_controller_test.rb:78:in `block in <class:UserSessionsControllerTest>'
```

**Comment by jywarren** [comment]
@SidharthBansal -- any ideas?

**Comment by SidharthBansal** [comment]
https://github.com/publiclab/plots2/issues/2880

**Comment by SidharthBansal** [comment]
@Souravirus is working on the pr related to it.
Thanks

**Comment by jywarren** [comment]
Try rebasing it?

**Comment by namangupta01** [comment]
Done 👍

**Comment by jywarren** [comment]
Awesome!

**Comment by jywarren** [comment]
Oh, huh. I see this error now on stable --

```
Building mailman
Cannot locate specified Dockerfile: Dockerfile
Makefile:2: recipe for target 'build' failed
```

Any ideas?

https://jenkins.laboratoriopublico.org/job/Plots-Stable/lastFailedBuild/console

**Comment by namangupta01** [comment]
is the docker path changed?

File: containers/docker-compose-production.yml [hunk 2]
```diff
@@ -21,3 +21,12 @@ services:
       - /etc/passwd:/etc/passwd:ro
       - /etc/group:/etc/group:ro
     restart: unless-stopped
+  mailman:
+    build: .
```
Comment (jywarren) [inline containers/docker-compose-production.yml]: maybe it has to be `..` here?

Comment (namangupta01) [reply containers/docker-compose-production.yml]: and in https://github.com/publiclab/plots2/blob/master/containers/docker-compose-production.yml#L20 also?

**Comment by jywarren** [comment]
hmm, maybe?

**Comment by namangupta01** [comment]
Made a pr #2885.

**Comment by jywarren** [comment]
Great!

**Comment by jywarren** [comment]
I believe so... do you mean are they set up with the password and email
address and server? If so, yes!

**Comment by namangupta01** [comment]
Yes, that's exactly what i mean.

**Pull request merged**